src/lib/proposer/proposer.ml

```diff
@@ -178,7 +180,20 @@ module Make (Inputs : Inputs_intf) :
               ( Ledger_builder_diff.With_valid_signatures_and_proofs.payments
                   diff
                 :> Payment.t list )
-            ~ledger:(Ledger_builder.ledger ledger_builder)
+            ~snarked_ledger_hash:(fun () ->
+              let open Or_error.Let_syntax in
+              match ledger_proof_opt with
+              | Some proof ->
+                  let snarked_ledger_hash =
+                    Ledger_proof.(statement proof |> statement_target) 
+                  in
+                  let%map snarked_ledger =
+                    Ledger_builder.snarked_ledger ledger_builder
+                      ~snarked_ledger_hash
+                  in
+                  Frozen_ledger_hash.of_ledger_hash
+                    (Ledger.merkle_root snarked_ledger)
+              | None -> Ok previous_ledger_hash )
```

Comment:
This piece of logic works under the assumption that previous_ledger_hash is the snarked ledger hash and not the staged ledger hash. However, this is not true right now. I think this may actually be a logic error in my code, not yours, but we need to fix this either way. 

https://github.com/CodaProtocol/coda/blob/master/src/lib/consensus/proof_of_stake.ml#L997